src/regexp.ts

```diff
@@ -103,12 +104,27 @@ export class BSONRegExp extends BSONValue {
     throw new BSONError(`Unexpected BSONRegExp EJSON object form: ${JSON.stringify(doc)}`);
   }
 
-  /** @internal */
-  [Symbol.for('nodejs.util.inspect.custom')](): string {
-    return this.inspect();
+  inspect(depth?: number, options?: unknown): string {
+    const stylize = getStylizeFunction(options);
+    const pattern = stylize(`'${this.pattern}'`, 'regexp');
+    const flags = stylize(`'${this.options}'`, 'regexp');
```

Comment:
Because of the way we build this string we will emit a broken string if `pattern` contains a single quote `'`, can we use inspect to get a properly quoted string and then stylize it?